source/math/FloatMatrix.ooc

```diff
@@ -113,9 +116,9 @@ FloatMatrix : cover {
 	// Creates a copy of the current matrix.
 	// </summary>
 	// <returns>Return a copy of the current matrix.</returns>
-	copy: func@ -> This {
-		result := This new(this dimensions)
-		memcpy(result elements data, this elements data, this dimensions area * Float size)
+	copy: func -> This {
+		result := This new(this _elements copy(), this dimensions)
+		this free(Owner Callee)
```

Comment:
It can be fixed later, I'm just wondering what's going on here, if there's something I'm missing. Why would you *ever* want to return a copy and free the original instead of just returning `this`?